webapp/channels/src/utils/burn_on_read_timer_ticker.ts

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+/**
+ * Global timer ticker for burn-on-read countdown displays.
+ * Uses a single self-correcting timer to broadcast tick events to all subscribers.
+ * This prevents O(n) intervals when displaying many burn-on-read posts.
+ *
+ * Features:
+ * - Self-correcting drift prevention (aligns to second boundaries)
+ * - Battery optimization (pauses when tab hidden)
+ * - Thread-safe snapshot iteration
+ * - Single timestamp per tick for perfect synchronization
+ */
+
+type TickCallback = (now: number) => void;
+
+class BurnOnReadTimerTicker {
+    private subscribers: Set<TickCallback> = new Set();
+    private timerId: ReturnType<typeof setTimeout> | null = null;
+    private readonly tickInterval = 1000; // 1 second
+
+    /**
+     * Subscribe to timer tick events
+     * Callback receives current timestamp for efficient time calculations
+     * Automatically starts the ticker if this is the first subscriber
+     */
+    public subscribe(callback: TickCallback): () => void {
+        this.subscribers.add(callback);
+
+        if (this.subscribers.size === 1) {
```

Comment:
Would it be more reliable to use a boolean `this.started` flag instead of size based check to avoid any race condition?